### ISSUE TITLE
Can't locate bad import

### ISSUE THREAD

**Issue by mjackson** [benmosher/eslint-plugin-import#379]
I'm pretty sure I don't have any `import`s for [this module](https://github.com/mjackson/web-starter/blob/master/modules/server/index.js), in [my project](https://github.com/mjackson/web-starter), but `import/default` complains about the lack of a `default` export. Is it possible to identify the offending `import` statement?

**Comment by benmosher** [comment]
The `default` rule flags import statements in which the _imported_ file has no default export. 

Which import is being reported?

**Comment by benmosher** [comment]
Ah, so: 
```
/Users/ben/git/web-starter/modules/server/index.js
  11:8  error  No default export found in module  import/default

✖ 1 problem (1 error, 0 warnings)
```
Means the import on line 11 has no default export, that line being:

```js
import webpackConfig from '../../webpack.config'
```

Since your webpack config is a CommonJS module, it has no default export, and thus the lint warning.

Things in `node_modules` are not expected to be ES6 by default, thus the lack of warnings for the other lines.

So: what would you expect to happen next? This is one of those weird boundary situations that I'm not sure how to handle (and thus, haven't yet). 

You could [explicitly ignore the file], at least short term. I'd love your perspective on what you'd expect the plugin to automatically do in this situation, though.

**Comment by benmosher** [comment]
Also, this is a similar situation to #268. I suspect the solution proposed there (attempt to detect CJS explicitly) would work here.

**Comment by mjackson** [comment]
Thanks for looking into this, @benmosher. My bad for not realizing the error message I needed was right in front of me the whole time.

This is an interesting edge case because I'm not even sure the ES6 module spec says anything about interop with CommonJS modules. I could be wrong, but I think babel just tries to do the right thing automatically.

Anyway, yes. I think detecting CommonJS would be a great start here. Then you could say something like "hey, you're using an ES6 import to import a CommonJS module. here are the rules around that". Of course, if I'm wrong and there is some spec that provides for importing CommonJS modules using ES6 imports then this whole thing may be a non-issue.

**Comment by mjackson** [comment]
I think for now what I'll do is just use `require` explicitly with CommonJS modules, since that's what it was designed for, and `import` with ES6 modules. That makes the intent a bit clearer too, IMO.
